### ISSUE TITLE
Request new Title and ID in dialog to generate a TD out of a TM

### ISSUE THREAD

**Issue by Silvanoc** [eclipse/editdor#76]
Thing Descriptions being generated out of Thing Models are keeping the `title` and `id` of the model. But typically you'll want different values for both fields. It would be nice to have to optional text boxes to define them in the "Generated TD from TM" dialog.

**Comment by sebastiankb** [comment]
The best way for initiating this is to use the placeholder concept. See the example here:

<img width="606" alt="image" src="https://user-images.githubusercontent.com/13832739/173085969-02f7f7a4-0231-4bca-8493-1411085d010a.png">

If you click on "Convert to TD" it will ask you to assign specific title and id for the TD instance.

**Comment by sebastiankb** [comment]
Just to avoid misunderstanding: The placeholder concept is a TM feature and not a ediTDor feature. Please see [here](https://w3c.github.io/wot-thing-description/#thing-model-td-placeholder).

**Comment by egekorkan** [comment]
I think that we should not make such assumptions, especially about the title since it is not meant to be unique. All Siemens iQ500 Washing Machines can have the title `Siemens iQ500 Washing Machine` that is displayed on a UI. In this case, the title for the TD can be already defined in the TM and never changed. 

`id` can be more up for discussion, i.e. whether it is allowed to conserve the `id` in TM->TD process. A more fundamental question is whether a TM should have an `id`. In the end, theoretically, `id` is for a Thing and not for a TD nor TM (https://w3c.github.io/wot-thing-description/#thing). In the TM stage, there should not even be an `id` unless there are placeholders, e.g. `"id": "urn:dev:ops:32473-iq500wm-{{serial_number}}",`. 

In any case, I would propose to move this discussion to the TD spec since editdor is not doing something wrong according to the TD spec.

**Comment by Silvanoc** [comment]
@egekorkan you're right with the assumptions WRT `title`. I see following options that might be valid use-cases and comply with the specification and should be somehow realizable with ediTDor:
* TM has a `title` different from that of the TDs.
* TM doesn't have any `title` and a title with placeholders can be used to be replaced on TD generation.
* TM has the same `title` as the generated TDs (pretty much what is mentioned in [your comment](https://github.com/eclipse/editdor/issues/76#issuecomment-1153308810) or in the [example I've mentioned above](https://www.w3.org/TR/wot-thing-description11/#example-62)).

WRT the `id`, IMO TMs should definitively have their own `id`s different from those of the TDs. Having a TM `id` with a placeholder would help differentiating it from the instantiated TDs, but would be somehow weird. IMO the way IDs are being used [example I've mentioned above](https://www.w3.org/TR/wot-thing-description11/#example-62) makes much more sense. And that's something ediTDor doesn't support as of now.
